File: app/src/main/java/com/booisajerk/activitycount/WorkoutHistory.java
package com.booisajerk.activitycount;

import android.os.Bundle;

import java.util.List;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProviders;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

public class WorkoutHistory extends AppCompatActivity {
    private ActivityRecordViewModel mActivityRecordViewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_history);

        RecyclerView recyclerView = findViewById(R.id.recyclerview);
        final ActivityRecordAdapter adapter = new ActivityRecordAdapter(this);
        recyclerView.setAdapter(adapter);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));

        mActivityRecordViewModel = ViewModelProviders.of(this).get(ActivityRecordViewModel.class);

        mActivityRecordViewModel.getAllActivityRecords().observe(this, new Observer<List<ActivityRecord>>() {
            @Override
            public void onChanged(@Nullable final List<ActivityRecord> records) {
                adapter.setActivityRecord(records);
            }
        });
    }
}

File: app/src/main/java/com/booisajerk/activitycount/WorkoutHistory.kt
package com.booisajerk.activitycount

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import kotlinx.android.synthetic.main.activity_history.*
import java.util.*

class WorkoutHistory : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_history)

        val adapter = WorkoutHistoryAdapter(generateSampleList())
        recyclerview.setHasFixedSize(true)
        recyclerview.layoutManager = LinearLayoutManager(this)
        recyclerview.adapter = adapter

        val dividerItemDecoration = DividerItemDecoration(recyclerview.getContext(),
                resources.configuration.orientation)
        recyclerview.addItemDecoration(dividerItemDecoration)
    }

    private fun generateSampleList(): List<WorkoutHistoryViewModel> {
        val viewModelList = ArrayList<WorkoutHistoryViewModel>()

        for (i in 9 downTo 1) {
            viewModelList.add(WorkoutHistoryViewModel(
                    String.format(Locale.US, "%d Aug 2018", i)
                    , String.format(Locale.US, "Push Ups: %d", i * i)
                    , String.format(Locale.US, "Squats: %d ", i + i)
                    , String.format(Locale.US, "Dips: %d ", i * i + i)))
        }

        return viewModelList
    }
}